src/StatusCodeSender.js

```diff
@@ -46,8 +22,10 @@ class StatusCodeSender {
 						case 504:
 							error.error = new Errors.GatewayTimeoutError();
 							break;
-					}
 
+						default:
+							error.error = new Errors.DefaultError(error.payload.errors[0].message)
```

Comment:
I think this could fail if `error.payload.errors[0].message` has any piece undefined. Could you add optional chaining to handle that case?

We should also add a test to handle that type of case and have a generic fallback error message if the error message comes through as undefined

- [ ]  Add test case for undefined message handling